### ISSUE TITLE
`reportDuplicateImport` false negative when duplicate import is in separate import statements

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#88]
pyright correctly reports `reportDuplicateImport` if the duplicated import is part of the same import statement:
```py
from typing import cast, cast # error: "cast" is imported more than once
```

but not when they are in separate import statements:
```py
from typing import cast
from typing import cast # no error
```

[playground](https://pyright-play.net/?reportDuplicateImport=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDGAhgM4wCwAUKJLIqhljnoafgMRQphQCmI4EEA)

rejected upstream issue: https://github.com/microsoft/pyright/issues/7261
